src/RobotSimulator/RobotSimulator.tsx

```diff
@@ -1,6 +1,68 @@
-import React, { FunctionComponent } from "react";
+import React, { Component } from "react";
 import "./RobotSimulator.css";
+import { Sim3D } from "./Sim3D";
 
-export const RobotSimulator: FunctionComponent = () => {
-  return <div className="robot-simulator">Robot Simulator</div>;
-};
+// This component coordinates between react html and the canvas. It uses the 3DSim class to handle the 3D scene and
+// proxies all required events from the browsers into the simulation. All react redux integration is done at this level.
+export class RobotSimulator extends Component {
+  private canvasRef = React.createRef<HTMLCanvasElement>();
+  private canvasParentRef = React.createRef<HTMLDivElement>();
+
+  private onresizeCb = this.onresize.bind(this);
+
+  private sim: Sim3D | null = null;
+  private isRendering = false;
+
+  componentDidMount() {
+    this.onresize();
+
+    window.addEventListener("resize", this.onresizeCb);
```

Comment:
Should we throttle or debounce this? Is resizing an expansive operation? See https://css-tricks.com/debouncing-throttling-explained-examples/.

TLDR: We can use `lodash/throttle` or `lodash/debounce` to only call the resize once after the user stops resizing the window or only X amount of times per second.

This can also be a follow up PR.

Reply:
lets make it a follow up, but i agree too